src/edge.rs

```diff
@@ -21,7 +23,18 @@ pub struct Edge {
 }
 
 impl Edge {
-    pub fn new(id: String, from_node: String, from_side: Option<Side>, from_end: Option<End>, to_node: String, to_side: Option<Side>, to_end: Option<End>, color: Option<crate::color::Color>, label: Option<String>) -> Edge {
+    #[allow(clippy::too_many_arguments)]
```

Comment:
I didn't know clippy lints, thanks!

Reply:
There are probably ways to refactor this so that clippy doesn't complain at all - for example, you could leave the color/ label out and expect people to set it afterwards. The other common way of handling with complicated setup like this would be to use the builder pattern https://rust-unofficial.github.io/patterns/patterns/creational/builder.html